### PR TITLE
dashboard revamp, #70

### PR DESCRIPTION
**Pull request by SunriseFox**

### CODE REVIEW

File: src/extension/options-page/NewPersonaCard.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Card from '@material-ui/core/Card'
+import { Link } from 'react-router-dom'
+import CardContent from '@material-ui/core/CardContent'
+import AddIcon from '@material-ui/icons/Add'
+import Button from '@material-ui/core/Button'
+import { Avatar } from '@material-ui/core'
+const useStyles = makeStyles(theme =>
+    createStyles({
+        card: {
+            minWidth: 375,
```
Comment (Jack-Works) [inline src/extension/options-page/NewPersonaCard.tsx]: on mobile?

Comment (neruthes) [reply src/extension/options-page/NewPersonaCard.tsx]: > on mobile?

iPhone 8 is 375. Should be small enough for 2019.

File: src/extension/options-page/Backup.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            <DialogTitle>{currentIdentifier ? currentIdentifier.friendlyToText() : 'Unknown'}</DialogTitle>
+            <DialogContent>
+                <div className={classes.container}>
+                    <Button onClick={downloadAsFile}>
+                        <a color="textPrimary">
+                            <Avatar className={classes.avatar}>
+                                <CloudDownloadIcon fontSize="large" />
+                            </Avatar>
+                        </a>
+                        <div>Download as File</div>
```
Comment (Jack-Works) [inline src/extension/options-page/Backup.tsx]: i18n

File: src/extension/options-page/NewPersonaCard.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    return (
+        <Card className={classes.card} raised elevation={3}>
+            <CardContent>
+                <Link component={Button} to={'/welcome'} className={classes.container}>
+                    <Avatar className={classes.avatar}>
+                        <AddIcon fontSize="large" />
+                    </Avatar>
+                </Link>
+                <div>Add New Persona</div>
```
Comment (Jack-Works) [inline src/extension/options-page/NewPersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+import { useColorProvider } from '../../utils/theme'
+
+interface Props {
+    identity: Person
+}
+
+const useStyles = makeStyles(theme =>
+    createStyles({
+        card: {
+            minWidth: 375,
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: on mobile

File: src/extension/options-page/PersonaCard.tsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+    useMemo(() => {
+        Services.Crypto.getMyProveBio(identity.identifier).then(p => setProvePost(p || ''))
+    }, [identity.identifier])
+
+    const [friendlyName, setFriendlyName] = useState<string>('')
+
+    useMemo(() => {
+        const ui = [...definedSocialNetworkUIs].find(i => i.networkIdentifier === identity.identifier.network)
+        setFriendlyName(ui ? ui.friendlyName : `Unknown(${identity.identifier.network})`)
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    }, [identity.identifier.network])
+
+    const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+
+    const copyPublicKey = () => {
+        navigator.clipboard
+            .writeText(provePost)
+            .then(() => {
+                enqueueSnackbar('Copied.', { variant: 'success', autoHideDuration: 1000 })
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+            })
+            .catch(e => {
+                enqueueSnackbar('Copy Failed.' + provePost, { variant: 'error' })
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    const undoDeleteIdentity = (restore: BackupJSONFileLatest) => {
+        const undo = (key: any) => {
+            Services.People.restoreBackup(restore)
+            closeSnackbar(key)
+        }
+        return (key: any) => (
+            <Button color="secondary" onClick={() => undo(key)}>
+                Undo
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+            .then(() => {
+                enqueueSnackbar('Copied.', { variant: 'success', autoHideDuration: 1000 })
+            })
+            .catch(e => {
+                enqueueSnackbar('Copy Failed.' + provePost, { variant: 'error' })
+            })
+    }
+
+    const undoDeleteIdentity = (restore: BackupJSONFileLatest) => {
+        const undo = (key: any) => {
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: prefer unknown than any

File: src/extension/options-page/PersonaCard.tsx [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+        )
+    }
+
+    const deleteIdentity = async () => {
+        const backup = await Services.Welcome.backupMyKeyPair(identity.identifier, {
+            download: false,
+            onlyBackupWhoAmI: true,
+        })
+        Services.People.removeMyIdentity(identity.identifier).then(() => {
+            enqueueSnackbar('Deleted.', { variant: 'default', action: undoDeleteIdentity(backup) })
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+        })
+    }
+
+    const [rename, setRename] = useState(false)
+
+    const renameIdentity = (event: React.FocusEvent<HTMLSpanElement> | React.KeyboardEvent<HTMLSpanElement>) => {
+        event.preventDefault()
+        Services.People.updatePersonInfo(identity.identifier, { nickname: event.currentTarget.innerText }).then(() => {
+            enqueueSnackbar('Done.', { variant: 'success', autoHideDuration: 1000 })
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+    return (
+        <Card className={classes.card} raised elevation={3}>
+            <CardContent>
+                <Typography className={classes.header} variant="h5" component="h2" gutterBottom>
+                    {
+                        <>
+                            <span
+                                suppressContentEditableWarning
+                                ref={titleRef}
+                                className={`title ${rename ? 'fullWidth' : ''}`}
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: use the classnames lib

File: src/extension/options-page/PersonaCard.tsx [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+    const titleRef = useRef<HTMLSpanElement | null>(null)
+
+    return (
+        <Card className={classes.card} raised elevation={3}>
+            <CardContent>
+                <Typography className={classes.header} variant="h5" component="h2" gutterBottom>
+                    {
+                        <>
+                            <span
+                                suppressContentEditableWarning
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: use Input instead

File: src/extension/options-page/PersonaCard.tsx [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+                </Typography>
+                <Typography className={classes.line} component="div">
+                    <div className="title" title={friendlyName}>
+                        {friendlyName}
+                    </div>
+                    <div className="content" title={identity.identifier.userId}>
+                        {identity.identifier.userId}
+                    </div>
+                    <div className="extra-item" hidden>
+                        Disconnect
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+                    </div>
+                </Typography>
+            </CardContent>
+            <Divider />
+            <CardContent>
+                <Typography className={classes.line} component="div">
+                    <div className="title">Public Key</div>
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+                    <div className="content" title={provePost}>
+                        {provePost}
+                    </div>
+                    <div className={`extra-item ${color.info}`} onClick={copyPublicKey}>
+                        Copy
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+                </Typography>
+            </CardContent>
+            <Divider />
+            <CardContent>
+                <Typography className={classes.line} component="div">
+                    <div className="title">Public Key</div>
+                    <div className="content" title={provePost}>
+                        {provePost}
+                    </div>
+                    <div className={`extra-item ${color.info}`} onClick={copyPublicKey}>
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: use classnames lib

File: src/extension/options-page/PersonaCard.tsx [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+            </CardContent>
+            <Divider />
+            <CardActions>
+                <Button
+                    size="small"
+                    className={color.info}
+                    color="primary"
+                    component={Link}
+                    to={`/backup?identity=${identity.identifier.toText()}`}>
+                    Create Backup
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/PersonaCard.tsx [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+                </Button>
+                <Button size="small" className={color.error} style={{ marginLeft: 'auto' }} onClick={deleteIdentity}>
+                    Delete Persona
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: i18n

File: src/extension/options-page/Welcome/index.tsx [hunk 20]
```diff
@@ -315,6 +302,7 @@ export default withRouter(function _WelcomePortal(props: RouteComponentProps) {
                     onFinish={() => {
                         if (webpackEnv.firefoxVariant === 'GeckoView' || webpackEnv.target === 'WKWebview')
                             window.close()
+                        enqueueSnackbar('Done.', { variant: 'success', autoHideDuration: 2000 })
```
Comment (Jack-Works) [inline src/extension/options-page/Welcome/index.tsx]: i18n

File: src/index.tsx [hunk 21]
```diff
@@ -1,193 +1,235 @@
...
-            width: drawerWidth,
+        actionButtons: {
+            margin: theme.spacing(2),
         },
-        content: {
-            flexGrow: 1,
-            width: '100%',
+        loaderWrapper: {
+            position: 'relative',
+            '&>MuiLinearProgress-root': {
```
Comment (Jack-Works) [inline src/index.tsx]: modify it with `classes`

File: src/index.tsx [hunk 22]
```diff
@@ -1,193 +1,235 @@
...
+    </>
+)
 
-    const isDarkTheme = useMediaQuery('(prefers-color-scheme: dark)')
-    const Router = (typeof window === 'object' ? HashRouter : MemoryRouter) as typeof HashRouter
+const ColorButton = withStyles((theme: Theme) => ({
+    root: {
+        color: theme.palette.getContrastText('#FFFFFF'),
+        padding: '0.5em 1.5rem',
+        boxShadow: '0px 0px 15px 1px rgba(0,0,0,0.1)',
```
Comment (Jack-Works) [inline src/index.tsx]: use pre defined shadow in theme

File: src/index.tsx [hunk 23]
```diff
@@ -1,193 +1,235 @@
...
+                            className={classes.logo}
+                            src="https://maskbook.com/img/maskbook--logotype-black.png"
+                            alt="Maskbook"
+                        />
+                        <Tabs
+                            value={currentTab}
+                            indicatorColor="primary"
+                            textColor="primary"
+                            onChange={handleTabChange}>
+                            <Tab label="Dashboard" />
```
Comment (Jack-Works) [inline src/index.tsx]: i18n

File: src/index.tsx [hunk 24]
```diff
@@ -1,193 +1,235 @@
...
+                        <section className={classes.cards}>
+                            {identities.map(i => (
+                                <PersonaCard identity={i} key={i.identifier.toText()} />
+                            ))}
+                            <NewPersonaCard />
+                        </section>
+                        <section className={classes.actionButtons}>
+                            <div className={classes.loaderWrapper}>
+                                {
+                                    // @ts-ignore
```
Comment (Jack-Works) [inline src/index.tsx]: avoid ts-ignore

File: src/index.tsx [hunk 25]
```diff
@@ -1,193 +1,235 @@
...
+                                    <Link onClick={exportData} to="" disabled={exportLoading} component={PaperButton}>
+                                        Export Backup Keystore
```
Comment (Jack-Works) [inline src/index.tsx]: i18n

File: src/index.tsx [hunk 26]
```diff
@@ -1,193 +1,235 @@
...
+                                    </Link>
+                                }
+                                {exportLoading && <LinearProgress />}
+                            </div>
+                            <Link to="/welcome?restore" component={PaperButton}>
+                                Import Data Backup
```
Comment (Jack-Works) [inline src/index.tsx]: i18n

File: src/index.tsx [hunk 27]
```diff
@@ -1,193 +1,235 @@
...
+                            </Link>
+                        </section>
+                    </main>
+                </Container>
+                <footer>
+                    <Breadcrumbs className={classes.footerButtons} separator="|" aria-label="breadcrumb">
+                        <FooterLink href="https://maskbook.com/">Maskbook.com</FooterLink>
+                        <FooterLink href="https://maskbook.com/privacy-policy/">Privacy Policy</FooterLink>
```
Comment (Jack-Works) [inline src/index.tsx]: i18n

File: src/utils/theme.ts [hunk 28]
```diff
@@ -21,8 +18,9 @@ function getFontFamily(monospace?: boolean) {
 const baseTheme = (theme: 'dark' | 'light') =>
     ({
         palette: {
-            primary: { main: indigo[400] },
+            primary: { main: '#2594ff' },
```
Comment (Jack-Works) [inline src/utils/theme.ts]: prefer mui color

File: src/utils/theme.ts [hunk 29]
```diff
@@ -36,8 +34,23 @@ const baseTheme = (theme: 'dark' | 'light') =>
...
+
 export const FixedWidthFonts = getFontFamily(true)
+
+export const useColorProvider = makeStyles((theme: typeof MaskbookDarkTheme) =>
+    createStyles({
+        error: {
+            color: theme.palette.error.main,
+        },
+        success: {
+            color: theme.palette.type === 'dark' ? green[800] : green[800],
```
Comment (Jack-Works) [inline src/utils/theme.ts]: same color?

File: src/extension/options-page/PersonaCard.tsx (hunk 13 above)
Comment (SunriseFox) [reply src/extension/options-page/PersonaCard.tsx]: don't know how to style Input the same as span

File: src/_locales/zh/messages.json [hunk 30]
```diff
@@ -105,6 +105,63 @@
...
+        "message": "公鑰"
+    },
+    "disconnect": {
+        "message": "斷開"
+    },
+    "dashboard": {
+        "message": "儀表板"
+    },
+    "synchronization": {
+        "message": "同步化"
```
Comment (Jack-Works) [inline src/_locales/zh/messages.json]: 🤔

File: src/_locales/zh/messages.json [hunk 31]
```diff
@@ -105,6 +105,63 @@
...
+        "message": "刪除身份"
+    },
+    "dashboard_show_qr": {
+        "message": "顯示二維碼"
+    },
+    "dashboard_hide_qr": {
+        "message": "隱藏二維碼"
+    },
+    "dashboard_item_copied": {
+        "message": "好了。"
```
Comment (Jack-Works) [inline src/_locales/zh/messages.json]: ???

File: src/extension/options-page/PersonaCard.tsx [hunk 32]
```diff
@@ -0,0 +1,232 @@
...
+    const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+
+    const copyPublicKey = () => {
+        navigator.clipboard
+            .writeText(provePost)
+            .then(() => {
+                enqueueSnackbar(geti18nString('dashboard_item_copied'), { variant: 'success', autoHideDuration: 1000 })
+            })
+            .catch(e => {
+                enqueueSnackbar(geti18nString('dashboard_item_copy_failed') + provePost, { variant: 'error' })
```
Comment (Jack-Works) [inline src/extension/options-page/PersonaCard.tsx]: maybe a prompt is more friendly to copy.

**Comment by SunriseFox** [comment]
@neruthes  There should be a close button in all dialogs and the page itself... or on mobile we are not able to close the dialogs or return to facebook.

File: src/extension/options-page/PersonaCard.tsx (hunk 13 above)
Comment (Jack-Works) [reply src/extension/options-page/PersonaCard.tsx]: https://github.com/DimensionDev/Maskbook/issues/355

File: src/_locales/zh/messages.json [hunk 33]
```diff
@@ -105,6 +105,63 @@
...
+    },
+    "dashboard_item_deleted": {
+        "message": "好了。"
+    },
+    "dashboard_item_done": {
+        "message": "好了。"
+    },
```
Comment (Misaka-0x447f) [inline src/_locales/zh/messages.json]: 您好了？

Comment (neruthes) [reply src/_locales/zh/messages.json]: 233

**Pull request merged**